prov/cxi/src/cxip_coll.c

```diff
@@ -3073,6 +3089,7 @@ static void _curl_delete_mc_obj(struct cxip_coll_mc *mc_obj)
 quit:
 	free(url);
 	if (ret < 0) {
+		mc_obj->close_state = ret;
```

Comment:
Here and below where you set mc_obj->close_state = ret and the call free(mc_obj). Why is this being done?

Reply:
Accessing memory after free. Works by chance but is undefined behavior. Fixing it.